system/Commands/Translation/LocalizationFinder.php

```diff
@@ -0,0 +1,341 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Commands\Translation;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use Config\App;
+use Locale;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use SplFileInfo;
+
+class LocalizationFinder extends BaseCommand
+{
+    protected $group       = 'Translation';
+    protected $name        = 'lang:find';
+    protected $description = 'Find and save available phrases to translate';
+    protected $usage       = 'lang:find [options]';
+    protected $arguments   = [];
+    protected $options     = [
+        '--locale'   => 'Apply changes to the selected locale (en, ru, etc...).',
+        '--dir'      => 'Directory for searching for translations (in relation to the APPPATH).',
+        '--show-new' => 'See only new translations as table.',
+        '--verbose'  => 'Output detailed information.',
+    ];
+
+    /**
+     * Flag for output detailed information
+     */
+    private bool $verbose = false;
+
+    private string $languagePath;
+
+    public function run(array $params)
+    {
+        $this->verbose      = array_key_exists('verbose', $params);
+        $cliOptionShowNew   = array_key_exists('show-new', $params);
+        $cliOptionLocale    = ! empty($params['locale']) ? $params['locale'] : null;
+        $cliOptionDir       = ! empty($params['dir']) ? $params['dir'] : null;
+        $currentLocale      = Locale::getDefault();
+        $currentDir         = APPPATH;
+        $this->languagePath = $currentDir . 'Language';
+        $tableRows          = [];
+        $languageFoundKeys  = [];
+        $countFiles         = 0;
+        $countNewKeys       = 0;
+
+        if (ENVIRONMENT === 'testing') {
+            $currentDir         = SUPPORTPATH . 'Services/';
+            $this->languagePath = SUPPORTPATH . 'Language/';
+        }
+
+        if (is_string($cliOptionLocale)) {
+            if (! in_array($cliOptionLocale, config(App::class)->supportedLocales, true)) {
+                CLI::error('Error: Supported locales ' . implode(', ', config(App::class)->supportedLocales));
+
+                return -1;
+            }
+
+            $currentLocale = $cliOptionLocale;
+        }
+
+        if (is_string($cliOptionDir)) {
+            $tempCurrentDir = realpath($currentDir . $cliOptionDir);
+
+            if (false === $tempCurrentDir) {
+                CLI::error('Error: Dir must be located in ' . $currentDir);
+
+                return -1;
+            }
+
+            if (str_starts_with($tempCurrentDir, $this->languagePath)) {
```

Comment:
`str_starts_with()` requires PHP 8.0+. We cannot use it now.

Reply:
symfony/polyfill not allowed?

Reply:
We do not use it.